src/pcre2_jit_compile.c

```diff
@@ -5687,11 +5714,44 @@ PCRE2_UCHAR othercase[1];
 #endif
 
 repeat = 1;
+stack_ptr = 0;
+step_count = 10000;
 while (TRUE)
   {
-  if (*rec_count == 0)
+  if (--step_count == 0)
     return 0;
-  (*rec_count)--;
+
+  SLJIT_ASSERT(chars <= chars_start + MAX_N_CHARS);
+
+  if (chars >= chars_end)
+    {
+    if (stack_ptr == 0)
+      return chars_end - chars_start;
```

Comment:
This `chars_end - chars_start` probably needs an explicit cast to int to silence warnings (even though we know it's always < 12 the compiler won't work it out always).